### ISSUE TITLE
Redundant records in the response

### ISSUE THREAD

**Issue by salzmdan** [NLnetLabs/nsd#50]
Hi,
I have noticed a regression in NSD 4.2.3. There are redundant records in the authority and additional sections in responses like:
```
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 6299
;; flags: qr aa rd; QUERY: 1, ANSWER: 1, AUTHORITY: 2, ADDITIONAL: 4
;; WARNING: recursion requested but not available

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags:; udp: 4096
;; QUESTION SECTION:
;ns2.zone10.			IN	AAAA

;; ANSWER SECTION:
ns2.zone10.		3600	IN	AAAA	2001:db8::2

;; AUTHORITY SECTION:
zone10.			3600	IN	NS	ns1.zone10.
zone10.			3600	IN	NS	ns2.zone10.

;; ADDITIONAL SECTION:
ns1.zone10.		3600	IN	A	192.0.2.1
ns2.zone10.		3600	IN	A	192.0.2.2
ns1.zone10.		3600	IN	AAAA	2001:db8::1
```
Relevant configuration:
```
minimal-responses: yes
```

**Comment by wcawijngaards** [comment]
Hi Daniel,
This issue looks to be fixed by #53 .  The new release contains a bug that is fixed in that change, the minimal responses code works, but the config is misread.

Thanks for the report.

**Comment by wcawijngaards** [comment]
That was merged, so this can be closed.  Also 4.2.4rc1 (a pre-release) is available for download should you want to upgrade to fix the problem.
